src/header/name.rs

```diff
@@ -2024,3 +2085,67 @@ fn test_eq_hdr_name() {
     let a = HeaderName { inner: Repr::Standard(Vary) };
     assert_ne!(a, b);
 }
+
+#[test]
+fn test_header_name_from_static() {
+    use std::panic;
+    use self::StandardHeader::Vary;
+
+    // Repr::Standard cases
+    let a = HeaderName { inner: Repr::Standard(Vary) };
+    
+    let b = HeaderName::from_static("vary");
+    assert_eq!(a, b);
+
+    let b = HeaderName::from_static("vaary");
+    assert_ne!(a, b);
+
+    let result = panic::catch_unwind(|| HeaderName::from_static("Vary"));
```

Comment:
What if each of these that should panic were separate `#[test] #[should_panic] fn foo() {}` test cases, so that `catch_unwind` wouldn't be needed? I think it might be clearer in reading. For example:

```rust
#[test]
#[should_panic]
fn from_static_uppercase() {
    HeaderName::from_static("Vary");
}
```

Since this would make a bunch of test functions, it might also be a good idea to push these into an inner `#[cfg(test)] mod tests { ... }`. Whatcha think?

Reply:
Definitely agree, and the other tests could also be moved into the block too.